### ISSUE TITLE
addon-import is generating files when it shouldn't

### ISSUE THREAD

**Issue by bcardarella** [ember-cli/ember-cli#4192]
For example:

A new addon. Run `ember g acceptance-test foo`

The `addon-import` generator will generate a `app/acceptance-tests/foo.js` file.

This is also happening for other test related file:

`ember g model-test foo` => `app/model-tests/foo.js`

**Comment by trabus** [comment]
I'll look into this.

**Comment by gmurphey** [comment]
Seems like the `addon-import` blueprint is being invoked for `*-test` blueprints.

**Comment by bcardarella** [comment]
I'm going to re-open this isue as #4217 did not completely address it. While it does address acceptance tests I specifically call out that the addon-import is affecting other test-related blueprints as well

**Comment by stefanpenner** [comment]
sorry the last close was due to 4217 auto-closing it.

@bcardarella if you have time to add a failing test-case that would be handy.

**Comment by trabus** [comment]
#4217 should stop test blueprints from generating an addon-import. I [added tests](https://github.com/ember-cli/ember-cli/pull/4217/files#diff-7c1a00c010ff3b072353212d766b1b8dR321) to the addon-generate-blueprint acceptance test to ensure that. 
If you'd also like an option to opt-out of generating the addon-import, I can do that as well.

**Comment by trabus** [comment]
Responding to this [comment](https://github.com/ember-cli/ember-cli/pull/4203#issuecomment-108242024)
> I think the issue is that the addon-import is an auto opt-in for all other blueprints. I think instead it should be done where addon-import must be specified on the blueprints that it should apply to.

@bcardarella Right now, the qualifier is if the blueprint has a `__root__` token (which allows it to be installed into an addon). We could change it to be a property of the blueprint that goes into the index.js, so it would be up to the blueprint on whether it generates the addon import.

We could also add an option to generate in an addon without generating an addon-import. Just spitballing, but maybe something like:
```
// from inside an addon, flag only
ember g component foo-bar --no-addon-import

// generates:
my-addon/addon/components/foo-bar.js
my-addon/addon/templates/components/foo-bar.js

// generate into app instead of addon folder
ember g component foo-bar --no-addon-import=app

// generates:
my-addon/app/components/foo-bar.js
my-addon/app/templates/components/foo-bar.js
```

**Comment by trabus** [comment]
@bcardarella I'm pretty sure #4217 does take care of this issue, I had a partial fix in #4203 that was then  superseded by #4217 with a proper fix that handled all test blueprints.

I'm also curious to know what you think about [this](https://github.com/ember-cli/ember-cli/issues/4192#issuecomment-108513023), and whether it would solve the auto opt-in problem you mentioned.
